### PR TITLE
BZ1973326: Update firstboot multipath doc

### PR DESCRIPTION
**Pull request by bobfuru**
[BZ1973326](https://bugzilla.redhat.com/show_bug.cgi?id=1973326)
Adds new procedure for day-1 enabling of multipath and updates wording for day 2 multipathing enablement. Fixes doc refs and wording to support this.

This PR relates to 4.8+ only. 

4.8 RN PR here: https://github.com/openshift/openshift-docs/pull/34341

**Preview links:**
- [Post-installation machine config tasks ("Enabling multipathing with kernel arguments on RHCOS")](https://deploy-preview-34462--osdocs.netlify.app/openshift-enterprise/latest/post_installation_configuration/machine-configuration-tasks.html#rhcos-enabling-multipath-day-2_post-install-machine-configuration-tasks)
- [Installing on bare metal UPI docs ("Installing RHCOS and starting the OpenShift Container Platform bootstrap process" --> "Enabling multipathing with kernel arguments on RHCOS")](https://deploy-preview-34462--osdocs.netlify.app/openshift-enterprise/latest/installing/installing_bare_metal/installing-bare-metal-network-customizations.html#rhcos-enabling-multipath_installing-bare-metal-network-customizations)

### CODE REVIEW

File: modules/rhcos-enabling-multipath.adoc [hunk 1]
```diff
@@ -1,119 +1,34 @@
...
-ip-10-0-141-105.ec2.internal   Ready,SchedulingDisabled   worker   28m   v1.20.0
-ip-10-0-142-249.ec2.internal   Ready                      master   34m   v1.20.0
-ip-10-0-153-11.ec2.internal    Ready                      worker   28m   v1.20.0
-ip-10-0-153-150.ec2.internal   Ready                      master   34m   v1.20.0
-----
-+
-You can see that scheduling on each worker node is disabled as the change is being applied.
 
 . Check that the kernel argument worked by going to one of the worker nodes and listing
 the kernel command line arguments (in `/proc/cmdline` on the host):
```
Comment (bobfuru) [inline modules/rhcos-enabling-multipath.adoc:34]: This is taken from the post-install multipath procedure. Is it relevant here or is there another way that the user can verify that the command worked?

**Comment by bobfuru** [comment]
Hi @jlebon - Could you PTAL? I'm unable to test currently from MacOS. I am hoping to run `coreos-installer` from a RHEL VM but in the meantime, I'm wondering if this is on the right track. Thanks!

File: modules/rhcos-enabling-multipath-day-2.adoc [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+// * post_installation_configuration/machine-configuration-tasks.adoc
+
+[id="rhcos-enabling-multipath-day-2_{context}"]
+= Enabling multipathing with kernel arguments on {op-system}
+
+{op-system} supports multipathing on the primary disk, allowing stronger resilience to hardware failure to achieve higher host availability. Post-installation support is available by activating multipathing via the machine config.
+
+[IMPORTANT]
+====
+Enabling multipathing during installation is supported and recommended for nodes provisioned in {product-title} 4.8 or higher. In setups where any I/O to non-optimized paths results in I/O system errors, you must enable multipathing at installation time.
```
Comment (jlebon) [inline modules/rhcos-enabling-multipath-day-2.adoc:12]: Should we link to the docs here where we discuss enabling it at installation time?

File: modules/rhcos-enabling-multipath.adoc (hunk 1 above)
Comment (jlebon) [reply modules/rhcos-enabling-multipath.adoc:34]: Yes, it's still relevant.

File: modules/rhcos-enabling-multipath.adoc [hunk 3]
```diff
@@ -1,119 +1,34 @@
...
-99-master-generated-registries                     52dd3ba6a9a527fc3ab42afac8d12b693534c8c9   3.2.0             33m
-99-master-ssh                                                                                 3.2.0             40m
-99-worker-generated-registries                     52dd3ba6a9a527fc3ab42afac8d12b693534c8c9   3.2.0             33m
-99-worker-ssh                                                                                 3.2.0             40m
-rendered-master-23e785de7587df95a4b517e0647e5ab7   52dd3ba6a9a527fc3ab42afac8d12b693534c8c9   3.2.0             33m
-rendered-worker-5d596d9293ca3ea80c896a1191735bb1   52dd3ba6a9a527fc3ab42afac8d12b693534c8c9   3.2.0             33m
-----
-
-. Check the nodes:
+. To enable multipathing at installation time, append the kernel arguments by entering the following command:
```
Comment (jlebon) [inline modules/rhcos-enabling-multipath.adoc:26]: There's two facets actually for a fully multipathed installation:
1. We want `coreos-installer install` to write to the multipathed device. This means that users should use e.g. `coreos-installer install /dev/mapper/mpatha` or using a WWN-named link under `/dev/disk/by-id`. That requires turning on multipath in the installation environment, which could be done using `rd.multipath=default` in the PXE/ISO boot, or just manually calling e.g. `mpathconf --enable` and `systemctl start multipathd.service` before calling `coreos-installer install`.
2. Appending kargs to the `coreos-installer install` command so that the installed system itself will use multipath right from the first boot. This part you already have.

File: modules/installation-complete-user-infra.adoc [hunk 4]
```diff
@@ -149,12 +149,17 @@ cluster machines.
 ifndef::ibm-power[]
 . For an installation with Fibre Channel Protocol (FCP), additional steps are required to enable multipathing. Do not enable multipathing during installation.
 +
-See "Enabling multipathing with kernel arguments on RHCOS" in the _Post-installation configuration_ documentation for more information.
+[NOTE]
+====
+When you install with multiple paths, you must enable multipathing directly after the installation, not at a later point in time, as this can cause problems.
```
Comment (jlebon) [inline modules/installation-complete-user-infra.adoc:154]: As per the previous comment, multipathing ideally would be turned on even before you do `coreos-installer install`, so it'd be more accurate to say "at installation time" instead of "after the installation". Maybe something like

```suggestion
When installing with multipath, it is strongly recommended to enable it at installation time, not at a later point in time, as this can cause problems.
```

?

File: modules/rhcos-enabling-multipath-day-2.adoc (hunk 2 above)
Comment (bobfuru) [reply modules/rhcos-enabling-multipath-day-2.adoc]: Yes, good call. Added.

File: modules/rhcos-enabling-multipath.adoc (hunk 3 above)
Comment (bobfuru) [reply modules/rhcos-enabling-multipath.adoc]: OK, thanks for the details. I've added an initial step based on whether you are doing an ISO/PXE boot or not. But I'm not sure if I should incorporate this part of your comment: 
>  This means that users should use e.g. `coreos-installer install /dev/mapper/mpatha` or using a WWN-named link under `/dev/disk/by-id`.

If so, how would you recommend doing so? As a prerequisite like the following perhaps?

```
.Prerequisites
* You are using either `coreos-installer install /dev/mapper/mpatha` or a World Wide Name (WWN) link under `/dev/disk/by-id`.
```

**Comment by bobfuru** [comment]
Thanks for reviewing, @jlebon. I've made some updates to address your comments, and left one question about whether to include your examples. 

Also, I realized I was missing  the procedure in a couple of the "Installing on bare metal" sections. It's now included in all three variations:
- Installing a user-provisioned cluster on bare metal
- Installing a user-provisioned bare metal cluster with network customizations
- Installing a user-provisioned bare metal cluster on a restricted network

PTAL, thanks!

**Comment by bobfuru** [comment]
@miabbott - PTAL for QE review, thank you!

File: modules/rhcos-enabling-multipath.adoc [hunk 5]
```diff
@@ -1,119 +1,54 @@
...
-. Create the new machine config by using either the master or worker YAML file you previously created:
+** For a PXE or ISO boot, add the `rd.multipath=default` kernel argument to your `MachineConfig` object file. For information about setting kernel arguments, see "Adding kernel arguments to nodes".
 +
-[source,terminal]
-----
-$ oc create -f ./99-master-kargs-mpath.yaml
-----
-
-. Check the machine configs to see that the new one was added:
+** For a non-PXE/ISO boot, run the following commands:
```
Comment (jlebon) [inline modules/rhcos-enabling-multipath.adoc:34]: How about something like:

- Turn on multipath using `mpathconf --enable && systemctl start multipathd.service` (for more information, see the [RHEL 8 multipath documentation](https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/8/html/configuring_device_mapper_multipath/assembly_setting-up-dm-multipath-configuring-device-mapper-multipath#proc_basic-multipath-setup-setting-up-dm-multipath)). If booting the PXE or ISO, you may also instead add `rd.multipath=default` to turn on multipath from the kernel command-line.
- This will allow you to target the multipath device directly for installation using e.g. `/dev/mapper/mpatha` or a WWN-named symlink in `/dev/disk/by-id`. As usual, the device path can be specified on the kernel command-line (i.e. via `coreos.inst.install_dev`) or interactively (i.e. when invoking `coreos-installer`).

? Not sure if it's too wordy.

Comment (bobfuru) [reply modules/rhcos-enabling-multipath.adoc]: @jlebon Made some more tweaks by incorporating the RHEL 8 docs. Thanks for the link. I need to make it task-based, so that's why it's broken into what seems like so many steps. Hoping I got the ordering and context right but please let me know. 🙇

File: modules/rhcos-enabling-multipath.adoc [hunk 6]
```diff
@@ -1,119 +1,50 @@
...
-----
+You can configure multipath so that the names of the multipath devices are of the form `/dev/mapper/mpath` `_n_`. Without this setting, the names of the multipath devices are aliased to a World Wide Name (WWN) symlink, such as `/dev/disk/by-label`.
 
-. Check the nodes:
+. Append the kernel arguments by entering the following command:
 +
 [source,terminal]
 ----
-$ oc get nodes
+$ coreos-installer install --append-karg rd.multipath=default --append-karg root=/dev/disk/by-label/dm-mpath-root
```
Comment (bobfuru) [inline modules/rhcos-enabling-multipath.adoc:46]: Is `by-label` sub dir okay here to stay in line with other examples in this section? Or should it really be `by-id`?

File: modules/rhcos-enabling-multipath.adoc [hunk 7]
```diff
@@ -1,119 +1,50 @@
...
-01-worker-kubelet                                  52dd3ba6a9a527fc3ab42afac8d12b693534c8c9   3.2.0             33m
-99-master-kargs-mpath                              52dd3ba6a9a527fc3ab42afac8d12b693534c8c9   3.2.0             105s
-99-master-generated-registries                     52dd3ba6a9a527fc3ab42afac8d12b693534c8c9   3.2.0             33m
-99-master-ssh                                                                                 3.2.0             40m
-99-worker-generated-registries                     52dd3ba6a9a527fc3ab42afac8d12b693534c8c9   3.2.0             33m
-99-worker-ssh                                                                                 3.2.0             40m
-rendered-master-23e785de7587df95a4b517e0647e5ab7   52dd3ba6a9a527fc3ab42afac8d12b693534c8c9   3.2.0             33m
-rendered-worker-5d596d9293ca3ea80c896a1191735bb1   52dd3ba6a9a527fc3ab42afac8d12b693534c8c9   3.2.0             33m
-----
+You can configure multipath so that the names of the multipath devices are of the form `/dev/mapper/mpath` `_n_`. Without this setting, the names of the multipath devices are aliased to a World Wide Name (WWN) symlink, such as `/dev/disk/by-label`.
```
Comment (jlebon) [inline modules/rhcos-enabling-multipath.adoc:40]: Unfortunately, we have no easy way right now for customers to provide custom multipath configs to the installed node on first boot. So they have to stay with the default configs emitted by `mpathconf --enable`.

If there's a single multipath device, then it *should* reliably be `/dev/mapper/mpatha`. If there are multiple multipath devices, or if they want to be more explicit, then using the `/dev/disk/by-id/wwn-<WWN>` symlinks would be best.

File: modules/rhcos-enabling-multipath.adoc (hunk 6 above)
Comment (jlebon) [reply modules/rhcos-enabling-multipath.adoc:46]: This path is good. Once RHCOS is installed, the root partition and label exist so it's valid. Before installing RHCOS, it's just a random disk, so they have to use e.g. `/dev/mapper/` or `/dev/disk/by-id/wwn-...` to refer to the multipathed disk when installing. Which, speaking of, should this example include the target device to be clearer? E.g.

```
coreos-installer install /dev/disk/by-id/wwn-0xx194e957fcedb4841 --append-karg rd.multipath=default --append-karg root=/dev/disk/by-label/dm-mpath-root
```

Comment (bobfuru) [reply modules/rhcos-enabling-multipath.adoc]: Sure, if that makes sense to you it's good with me. 😁  I changed the lead-in wording and the code block slightly to indicate that this is an example with a replaceable value. Do my edits make sense?

File: modules/rhcos-enabling-multipath.adoc (hunk 7 above)
Comment (bobfuru) [reply modules/rhcos-enabling-multipath.adoc]: Forgive my ignorance, please, as I try to understand the steps that the user would take here.

Can you provide an example output that the user would see after running step 1 (`$ mpathconf --enable && systemctl start multipathd.service`)? I'm thinking that could help with step two, where I've changed it to reflect your wording by saying, "Use the default configs emitted by `mpathconf --enable`".

File: modules/rhcos-enabling-multipath.adoc [hunk 8]
```diff
@@ -1,122 +1,57 @@
...
+** Optional: If booting the PXE or ISO, you can instead enable multipath by adding `rd.multipath=default` from the kernel command line.
 
-. Check the machine configs to see that the new one was added:
+. Use the default configs emitted by `mpathconf --enable` from the previous step to specify the multipath device directly for installation, either on the kernel command line by using `coreos.inst.install_dev` or interactively by invoking the `coreos-installer` program.
 +
-[source,terminal]
-----
-$ oc get MachineConfig
-----
+For a single multipath device, use the path `/dev/mapper/mpatha`.
```
Comment (bobfuru) [inline modules/rhcos-enabling-multipath.adoc:40]: Is there a `coreos-installer` command example we could provide here?

File: modules/rhcos-enabling-multipath.adoc (hunk 7 above)
Comment (bobfuru) [reply modules/rhcos-enabling-multipath.adoc]: Thanks, @jlebon, and pardon my ignorance in this process, please. 😁 

I've made another round of updates to hopefully give clearer directions on what users need to do to append kargs using `coreos-installer install`. So now in Step 2, I've broken it out for instructions to do either: A) A single mpath device, or B) Multiple mpath devices. I also left a note in there that they can use `coreos.inst.install_dev`, though I'm not certain how. If you have a suggestion for what step(s) the user should take for this, let me know and I'll include it. Thanks!

File: modules/rhcos-enabling-multipath.adoc [hunk 9]
```diff
@@ -1,122 +1,65 @@
...
-99-master-kargs-mpath                              52dd3ba6a9a527fc3ab42afac8d12b693534c8c9   3.2.0             105s
-99-master-generated-registries                     52dd3ba6a9a527fc3ab42afac8d12b693534c8c9   3.2.0             33m
-99-master-ssh                                                                                 3.2.0             40m
-99-worker-generated-registries                     52dd3ba6a9a527fc3ab42afac8d12b693534c8c9   3.2.0             33m
-99-worker-ssh                                                                                 3.2.0             40m
-rendered-master-23e785de7587df95a4b517e0647e5ab7   52dd3ba6a9a527fc3ab42afac8d12b693534c8c9   3.2.0             33m
-rendered-worker-5d596d9293ca3ea80c896a1191735bb1   52dd3ba6a9a527fc3ab42afac8d12b693534c8c9   3.2.0             33m
+$ coreos-installer install /dev/mapper/mpatha \ <1>
+--append-karg rd.multipath=default \
+--append-karg root=/dev/mapper/dm-mpath-root
```
Comment (bobfuru) [inline modules/rhcos-enabling-multipath.adoc:46]: Please double-check if these paths are correct for this example.

File: modules/rhcos-enabling-multipath.adoc [hunk 10]
```diff
@@ -1,122 +1,65 @@
...
-ip-10-0-153-11.ec2.internal    Ready                      worker   28m   v1.20.0
-ip-10-0-153-150.ec2.internal   Ready                      master   34m   v1.20.0
+$ coreos-installer install /dev/disk/by-id/<wwn_ID> \ <1>
+--append-karg rd.multipath=default \
+--append-karg root=/dev/disk/by-label/dm-mpath-root
 ----
+<1> Indicates the WWN ID of the target multipathed device. For example, `wwn-0xx194e957fcedb4841`.
 +
-You can see that scheduling on each worker node is disabled as the change is being applied.
+Alternatively, you can append the kernel arguments on the kernel command line by using `coreos.inst.install_dev`.
```
Comment (bobfuru) [inline modules/rhcos-enabling-multipath.adoc:60]: Should we describe how this is done as an optional step?

File: modules/rhcos-enabling-multipath.adoc (hunk 9 above)
Comment (jlebon) [reply modules/rhcos-enabling-multipath.adoc:46]:
```suggestion
$ coreos-installer install /dev/mapper/mpatha \ <1>
--append-karg rd.multipath=default \
--append-karg root=/dev/disk/by-label/dm-mpath-root
```

The `root=` karg should always be `/dev/disk/by-label/dm-mpath-root`. Here we're talking about the path to the overall block device itself onto which all of RHCOS is installed (e.g. on top of which the boot partition, EFI and rootfs partitions are created).

File: modules/rhcos-enabling-multipath.adoc [hunk 11]
```diff
@@ -1,122 +1,65 @@
...
 ----
-
-. Check the nodes:
+<1> Indicates the path of the single multipathed device.
 +
-[source,terminal]
-----
-$ oc get nodes
-----
+* For multiple multipath devices, or to be more explicit, it is recommended that the names of the multipath devices are aliased to a World Wide Name (WWN) symlink when you append the kernel arguments , such as `/dev/disk/by-label`. For example:
```
Comment (jlebon) [inline modules/rhcos-enabling-multipath.adoc:50]:
```suggestion
* For multiple multipath devices, or to be more explicit, instead of using e.g. `/dev/mapper/mpatha`, it is recommended to use the World Wide Name (WWN) symlink available in `/dev/disk/by-id`. For example:
```

File: modules/rhcos-enabling-multipath.adoc [hunk 12]
```diff
@@ -1,122 +1,65 @@
...
 [source,terminal]
 ----
-NAME                           STATUS                     ROLES    AGE   VERSION
-ip-10-0-136-161.ec2.internal   Ready                      worker   28m   v1.20.0
-ip-10-0-136-243.ec2.internal   Ready                      master   34m   v1.20.0
-ip-10-0-141-105.ec2.internal   Ready,SchedulingDisabled   worker   28m   v1.20.0
-ip-10-0-142-249.ec2.internal   Ready                      master   34m   v1.20.0
-ip-10-0-153-11.ec2.internal    Ready                      worker   28m   v1.20.0
-ip-10-0-153-150.ec2.internal   Ready                      master   34m   v1.20.0
+$ coreos-installer install /dev/disk/by-id/<wwn_ID> \ <1>
```
Comment (jlebon) [inline modules/rhcos-enabling-multipath.adoc:54]: Should this be

```suggestion
$ coreos-installer install /dev/disk/by-id/wwn-<wwn_ID> \ <1>
```

instead, and then the `<1>` below is just `0xx194e957fcedb4841` for the WWN ID?

File: modules/rhcos-enabling-multipath.adoc (hunk 10 above)
Comment (jlebon) [reply modules/rhcos-enabling-multipath.adoc:60]: Yeah, not sure where it belongs. Maybe better to just link to the docs where we detail the "install via coreos.inst kargs" path? E.g. something like:

> This symlink can also be used as the `coreos.inst.install_dev` kernel argument when using the [automated installation procedure](link to relevant page).

?

File: modules/rhcos-enabling-multipath.adoc (hunk 12 above)
Comment (bobfuru) [reply modules/rhcos-enabling-multipath.adoc]: Resolved

File: modules/rhcos-enabling-multipath.adoc (hunk 11 above)
Comment (bobfuru) [reply modules/rhcos-enabling-multipath.adoc]: Resolved

File: modules/rhcos-enabling-multipath.adoc (hunk 10 above)
Comment (bobfuru) [reply modules/rhcos-enabling-multipath.adoc]: Sure, that sounds good. We currently have technical limitations to our docs that don't allow xrefs from a module such as this, but I have referenced the title of the relevant doc section.

File: modules/rhcos-enabling-multipath.adoc (hunk 9 above)
Comment (bobfuru) [reply modules/rhcos-enabling-multipath.adoc]: Thanks for the explanation, resolved.

**Comment by bobfuru** [comment]
Thanks for the additional feedback, @jlebon. Please let me know if I've missed anything or if this looks good.

File: modules/rhcos-enabling-multipath.adoc [hunk 13]
```diff
@@ -1,122 +1,65 @@
...
+** Optional: If booting the PXE or ISO, you can instead enable multipath by adding `rd.multipath=default` from the kernel command line.
 
-. Check the machine configs to see that the new one was added:
+. Append the kernel arguments by invoking the `coreos-installer` program:
 +
-[source,terminal]
-----
-$ oc get MachineConfig
-----
+* For a single multipath device, use the path `/dev/mapper/mpatha`, for example:
```
Comment (jlebon) [inline modules/rhcos-enabling-multipath.adoc:40]: Maybe instead, something like:

```suggestion
* If there is only one multipath device connected to the machine, it should be available at path `/dev/mapper/mpatha`. For example:
```

File: modules/rhcos-enabling-multipath.adoc [hunk 14]
```diff
@@ -1,122 +1,65 @@
...
 ----
-
-. Check the nodes:
+<1> Indicates the path of the single multipathed device.
 +
-[source,terminal]
-----
-$ oc get nodes
-----
+* For multiple multipath devices or to be more explicit, instead of using a path such as `/dev/mapper/mpatha`, it is recommended to use the World Wide Name (WWN) symlink available in `/dev/disk/by-id`. For example:
```
Comment (jlebon) [inline modules/rhcos-enabling-multipath.adoc:50]: And then here

```suggestion
* If there are multiple multipath devices or to be more explicit, instead of using `/dev/mapper/mpatha`, it is recommended to use the World Wide Name (WWN) symlink available in `/dev/disk/by-id`. For example:
```

**Review by jlebon — APPROVED** [review]
Minor wording tweaks, otherwise LGTM!

File: modules/rhcos-enabling-multipath.adoc (hunk 14 above)
Comment (bobfuru) [reply modules/rhcos-enabling-multipath.adoc]: Resolved

File: modules/rhcos-enabling-multipath.adoc (hunk 13 above)
Comment (bobfuru) [reply modules/rhcos-enabling-multipath.adoc]: Resolved

**Comment by bobfuru** [comment]
Thanks, @jlebon - made those last couple of tweaks and will get this merged at 4.8 GA. 😀

**Comment by vikram-redhat** [comment]
@bobfuru does this need to be merged now?

**Comment by bobfuru** [comment]
If possible, yes @vikram-redhat - after peer review. Thanks.

File: modules/rhcos-enabling-multipath-day-2.adoc [hunk 15]
```diff
@@ -0,0 +1,139 @@
...
+====
+// Add xref once it's allowed.
+
+.Prerequisites
+* You have a running {product-title} cluster that uses version 4.7 or later.
+* You are logged in to the cluster as a user with administrative privileges.
+
+.Procedure
+
+. To enable multipathing post-installation on control plane nodes (also known as the master nodes):
```
Comment (stevsmit) [inline modules/rhcos-enabling-multipath-day-2.adoc:27]: Is there a steadfast rule on parentheses? I've always been told to steer away from them.

File: modules/rhcos-enabling-multipath-day-2.adoc [hunk 16]
```diff
@@ -0,0 +1,139 @@
...
+// * post_installation_configuration/machine-configuration-tasks.adoc
+
+[id="rhcos-enabling-multipath-day-2_{context}"]
+= Enabling multipathing with kernel arguments on {op-system}
+
+{op-system} supports multipathing on the primary disk, allowing stronger resilience to hardware failure to achieve higher host availability. Post-installation support is available by activating multipathing via the machine config.
+
+[IMPORTANT]
+====
+Enabling multipathing during installation is supported and recommended for nodes provisioned in {product-title} 4.8 or higher. In setups where any I/O to non-optimized paths results in I/O system errors, you must enable multipathing at installation time. For more information about enabling multipathing during installation time, see "Enabling multipathing with kernel arguments on RHCOS" in the _Installing on bare metal_ documentation.
```
Comment (stevsmit) [inline modules/rhcos-enabling-multipath-day-2.adoc:12]: Would it be {product-title} {product-version} since it's for 4.8?

File: modules/rhcos-enabling-multipath.adoc [hunk 17]
```diff
@@ -1,122 +1,65 @@
...
 [id="rhcos-enabling-multipath_{context}"]
 = Enabling multipathing with kernel arguments on {op-system}
 
 {op-system} supports multipathing on the primary disk, allowing stronger resilience to hardware failure to achieve higher host availability.
 
-[IMPORTANT]
-====
-Multipathing is only supported when it is activated using the machine config as documented in the following procedure. It must be enabled after {op-system} installation.
-====
+You can enable multipathing at installation time for nodes that were provisioned in {product-title} 4.8 or later. While post-installation support is available by activating multipathing via the machine config, enabling multipathing during installation is recommended.
```
Comment (stevsmit) [inline modules/rhcos-enabling-multipath.adoc:12]: {product-version} ?

File: modules/rhcos-enabling-multipath-day-2.adoc [hunk 18]
```diff
@@ -0,0 +1,139 @@
+// Module included in the following assemblies:
+//
+// * post_installation_configuration/machine-configuration-tasks.adoc
+
+[id="rhcos-enabling-multipath-day-2_{context}"]
+= Enabling multipathing with kernel arguments on {op-system}
```
Comment (stevsmit) [inline modules/rhcos-enabling-multipath-day-2.adoc:6]: There is no {op-system-first} on this entire page. Not sure if that matters, though.

File: installing/installing_bare_metal/installing-bare-metal.adoc [hunk 19]
```diff
@@ -32,7 +32,7 @@ include::modules/cluster-entitlements.adoc[leveloffset=+1]
 
 .Additional resources
```
[the remark this answers is not in the document]
Comment (bobfuru) [reply installing/installing_bare_metal/installing-bare-metal.adoc:33]: It's convoluted but in cases such as this one, we don't use a unique ID because we want it to display as a "[trailing include](https://github.com/openshift/openshift-docs/blob/master/contributing_to_docs/doc_guidelines.adoc#:~:text=Only%20use%20.%20syntax%20(.Additional%20resources)%20to%20follow%20the%20module%20as%20a%20trailing%20include%20at%20the%20assembly%20level%2C%20which%20indicates%20that%20the%20resources%20apply%20specifically%20to%20the%20module.)" that relates to the preceding module rather than to the assembly as a whole.

> **Note**
> When using "Prerequisites", "Next steps", and "Additional resources" headings in an assembly, use == formatting, such as == Prerequisites and == Additional resources. Use of this heading syntax at the assembly level indicates that the sections relate to the whole assembly. Only use . syntax (.Additional resources) to follow the module as a trailing include at the assembly level, which indicates that the resources apply specifically to the module.

File: modules/rhcos-enabling-multipath-day-2.adoc (hunk 15 above)
Comment (bobfuru) [reply modules/rhcos-enabling-multipath-day-2.adoc:27]: Good catch. With 4.7, I believe we were using this boilerplate wording that was using parentheses. I've gone ahead and removed it here because we are no longer using the "aka" wording starting with 4.8. (And yes, you're right, we try to follow ISG's suggestion to avoid parentheses by rewriting or by using commas instead. 😁 )

File: modules/rhcos-enabling-multipath-day-2.adoc (hunk 16 above)
Comment (bobfuru) [reply modules/rhcos-enabling-multipath-day-2.adoc:12]: Here, we want to hardcode the version to be sure that users know it's a feature that started in 4.8. In future versions, they'll want to know that they can use 4.8+ nodes, as opposed to {product-version}+ nodes.

File: modules/rhcos-enabling-multipath.adoc (hunk 17 above)
Comment (bobfuru) [reply modules/rhcos-enabling-multipath.adoc:12]: Same comment as above, leaving 4.8 hardcoded.

File: modules/rhcos-enabling-multipath-day-2.adoc (hunk 18 above)
Comment (bobfuru) [reply modules/rhcos-enabling-multipath-day-2.adoc:6]: Yes, I think we should use {op-system-first} on first instance here in the paragraph text. Thanks!

File: modules/rhcos-enabling-multipath.adoc [hunk 20]
```diff
@@ -1,122 +1,65 @@
...
-ip-10-0-153-11.ec2.internal    Ready                      worker   28m   v1.20.0
-ip-10-0-153-150.ec2.internal   Ready                      master   34m   v1.20.0
+$ coreos-installer install /dev/disk/by-id/wwn-<wwn_ID> \ <1>
+--append-karg rd.multipath=default \
+--append-karg root=/dev/disk/by-label/dm-mpath-root
 ----
+<1> Indicates the WWN ID of the target multipathed device. For example, `0xx194e957fcedb4841`.
 +
-You can see that scheduling on each worker node is disabled as the change is being applied.
+This symlink can also be used as the `coreos.inst.install_dev` kernel argument when using special `coreos.inst.*` arguments to direct the live installer. For more information, see "Creating {op-system-first} machines".
```
[the remark this answers is not in the document]
Comment (bobfuru) [reply modules/rhcos-enabling-multipath.adoc:60]: Definitely! It also made me realize that we changed the title of the xref in 4.8, so updated that here as well. Thanks!

**Comment by bobfuru** [comment]
Thanks for the review, @stevsmit!
@vikram-redhat this is ready for your ack/approval/merge. 🙏

**Review by vikram-redhat — APPROVED** [review]
